packages/api/src/controllers/animalController.js

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalModel from '../models/animalModel.js';
+import baseController from './baseController.js';
+
+const animalController = {
+  getFarmAnimals() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalModel.query().where({ farm_id });
+        return res.status(200).send(rows);
+      } catch (error) {
+        console.error(error);
+        return res.status(500).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addAnimal() {
+    return async (req, res) => {
+      try {
+        const trx = await transaction.start(Model.knex());
+        const { farm_id } = req.headers;
+        const result = [];
+
+        if (!Array.isArray(req.body)) {
+          return res.status(400).send('Request body should be an array');
+        }
+
+        for (const animal of req.body) {
+          if (!animal.identifier && !animal.name) {
+            return res.status(400).send('Should send either animal name or identifier');
+          }
+
+          if (!animal.default_breed_id && !animal.custom_breed_id) {
+            return res.status(400).send('Should send either default_breed_id or custom_breed_id');
+          }
+
+          const individualAnimalResult = await baseController.postWithResponse(
+            AnimalModel,
+            { ...animal, farm_id },
```

Comment:
You did this correctly -- but if for some reason farm_id came before ...animal then it would be possible to override farm_id and I could add animals to another persons farm accidentally or on purpose. 

There is some risk that some developer who copies this pattern in the future does not know/remember that -- it could be prudent to either only spread specific animal properties or check the body against farm_id.

Reply:
Yeah, I could just strip farm_id off of the animal object before spreading to make extra sure

Reply:
It might be easier to specify what we want. There could be other table columns we don't want the user to specify in their body req to addAnimals. And instead let the logic do its thing.

Not sure this is possible but user_id, deleted, id etc.  <<- I havent tested that yet though

Reply:
@antsgar Yes you switched the farm_id so that solves cross farm issues. 

But the principle applies to the baseProperties too. So I could add a created_by_user_id="someones_id_I_want_to_frame_bad_data" or deleted=true on creation. 

I might be wrong on the importance here but seems like something we want the context and defaults to handle.